### ISSUE TITLE
[BUG] Playwright-chromium (>1.5.0) does not exit properly (hangs) when started in docker

### ISSUE THREAD

**Issue by DJ-Glock** [microsoft/playwright#4218]
**Context:**
- Playwright Version: playwright-chromium 1.5.0, 1.5.1 (in 1.4.0 everything was fine).
- Operating System: Windows 10/MacOS Catalina + Docker
- Node.js version: 12.18.3
- Browser: Chromium
- Extra: Docker image based on PW:bionic; mocha for running tests;

**Bug description:**
Playwright hangs after tests execution when running in Docker. It simply does not do anything (as per logs).

**Repro steps:**
1. Unpack sample project, go to root folder
2. Run the following command:
`docker-compose up --build --abort-on-container-exit --exit-code-from e2e > test_run.log`
3. Check test_run.log. Once all tests are finished, you will see the following lines in log:

> [36me2e_1  |[0m ##teamcity[testFinished name='Test_1 Test 5 - UI Open Search page, find user and open it - chromium' duration='7284' flowId='28']
> [36me2e_1  |[0m ##teamcity[testSuiteFinished name='Test_1 Google and Twitter checks - chromium' duration='' flowId='28']

But the process will stay alive and docker container will never exit (at least in 2 hours).
[qa-automation.zip](https://github.com/microsoft/playwright/files/5430155/qa-automation.zip)

I was also able to reproduce this behaviour on my Mac using provided test project.

**Comment by pavelfeldman** [comment]
Browser can't close gracefully and is not killed on failure. Investigating.

```
<gracefully close start>
Received signal 11 SEGV_MAPERR ffffed3d1aef8ebd
#0 0x564a1fa6d2f9 base::debug::CollectStackTrace()
#1 0x564a1f9e3743 base::debug::StackTrace::StackTrace()
#2 0x564a1fa6ce9b base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f36e587b8a0 (/lib/x86_64-linux-gnu/libpthread-2.27.so+0x1289f)
#4 0x564a1e05d565 content::protocol::NetworkHandler::SetNetworkConditions()
#5 0x564a1e05d4b6 content::protocol::NetworkHandler::Disable()
#6 0x564a1e02ff86 content::DevToolsSession::Dispose()
#7 0x564a1e01a78e content::DevToolsAgentHostImpl::DetachInternal()
#8 0x564a1e01a68d content::DevToolsAgentHostImpl::DetachClient()
#9 0x564a1e0884d4 content::protocol::TargetHandler::Session::Detach()
#10 0x564a1e08236e content::protocol::TargetAutoAttacher::ReattachTargetsOfType()
#11 0x564a1e08388f content::protocol::TargetAutoAttacher::SetAutoAttach()
#12 0x564a1e087b34 content::protocol::TargetHandler::SetAutoAttachInternal()
#13 0x564a1e0878c3 content::protocol::TargetHandler::Disable()
#14 0x564a1e02ff86 content::DevToolsSession::Dispose()
#15 0x564a1e01a78e content::DevToolsAgentHostImpl::DetachInternal()
#16 0x564a1e01a68d content::DevToolsAgentHostImpl::DetachClient()
#17 0x564a1e02cbe2 content::DevToolsPipeHandler::Shutdown()
#18 0x564a1e02cedc content::DevToolsPipeHandler::~DevToolsPipeHandler()
#19 0x564a24968a94 headless::HeadlessBrowserMainParts::PostMainMessageLoopRun()
#20 0x564a1dfa41ec content::BrowserMainLoop::ShutdownThreadsAndCleanUp()
#21 0x564a1dfa5f47 content::BrowserMainRunnerImpl::Shutdown()
#22 0x564a24974e1b headless::HeadlessContentMainDelegate::RunProcess()
#23 0x564a1f98e907 content::ContentMainRunnerImpl::RunServiceManager()
#24 0x564a1f98e53e content::ContentMainRunnerImpl::Run()
#25 0x564a1f98b956 content::RunContentProcess()
#26 0x564a1f98c33c content::ContentMain()
#27 0x564a1f9dd822 headless::(anonymous namespace)::RunContentMain()
#28 0x564a1f9dd50c headless::HeadlessShellMain()
#29 0x564a1d1a0883 ChromeMain
#30 0x7f36dfde6b97 __libc_start_main
#31 0x564a1d1a06aa _start
  r8: 0000000000000000  r9: 0000000000000000 r10: 000000000000005f r11: 0000000000000000
 r12: 000012c0572e12d0 r13: 000012c057771e60 r14: 00007ffcbe2d43e0 r15: 000012c0572e12d0
  di: 000012c056fd5800  si: 00007ffcbe2d43e0  bp: 00007ffcbe2d43d0  bx: 0000000000000000
  dx: 000012c0573a7380  ax: ffffed3d1aef8eb5  cx: aaaaaaaaaaaaaaaa  sp: 00007ffcbe2d43a0
  ip: 0000564a1e05d565 efl: 0000000000010206 cgf: 002b000000000033 erf: 0000000000000005
 trp: 000000000000000e msk: 0000000000000000 cr2: ffffed3d1aef8ebd
```

**Comment by pavelfeldman** [comment]
Offending patch: https://github.com/microsoft/playwright/pull/3931

**Comment by pavelfeldman** [comment]
Still need to fix that Chromium crash!

**Comment by pavelfeldman** [comment]
Reduced test case:
```
await page.goto('https://twitter.com');
```

**Comment by pavelfeldman** [comment]
The hang is fixed and will be released with 1.5.2. shortly. The Chromium issue is about service workers, is now filed upstream and I'm working on a fix there.

**Comment by pirate** [comment]
Can you provide the link to your upsteam chromium issue? @pavelfeldman I'm investigating a similar issue around chrome hanging on exit and am curious to see how yours was resolved.
